### ISSUE TITLE
Improve implementation of open solvers for SBGrid

### ISSUE THREAD

**Issue by stephaniewankowicz** [ExcitedStates/qfit-3.0#374]
We want a set of working open solvers to be able to put qFit into SBGrid

**Comment by blake-riley** [comment]
IMO, this was done in #363, which reimplemented cvxopt (QP), osqp (QP) and miosqp (MIQP).
Or is this just an issue to track until dev is merged with main?

**Comment by stephaniewankowicz** [comment]
The open MIQP solver does not work currently (https://github.com/osqp/miosqp)

**Comment by blake-riley** [comment]
It crashes? Doesn't load?
Can you give me a crash log, and I'll look into it tonight?

**Comment by stephaniewankowicz** [comment]
This issue is with the following line:

class MIOSQPSolver(MIQPSolver):
    driver_pkg_name = "miosqp"
    driver = lazy_load_module_if_available(driver_pkg_name)
    if TYPE_CHECKING:
        import miosqp

The import fails. This package is not included in osqp. (https://github.com/osqp/miosqp)

**Comment by blake-riley** [comment]
That's correct. `miosqp` is a separate package.
Unfortunately though, the authors haven't distributed it in the pip repositories, only on GitHub.

It's not installing for 2 reasons:

1. I made a mistake in `setup.py`. Where it says `extra_dependencies`, it should read `extras_require`.
2. Even if that was fixed: it's not obvious, but when you install qfit with `pip install .` (or `pip install qfit` one day, I guess), you'd have to specify the _set_ of qfit "extensions" you'd like in square-brackets (`pip install .[osqp]` or `pip install .[cplex,osqp]`).[^cplex]

Since we need to have at least one set of solvers, we should probably make the osqp solvers _default_, and just include them straight up in the `install_requires`. This would satisfy SBGrid.

I will put in a PR to fix this tomorrow morning?

[^cplex]: Even then, cplex is only _partially_ available through pip, so this only really works on old macs, or on linux machines.
